internal/common/vote/vote.go

```diff
@@ -0,0 +1,191 @@
+package vote
+
+import (
+	"math/rand"
+	"time"
+)
+
+func GetVotesForRule(ruleID string, islandID int) map[int][]int {
+	votes_layout := make(map[int][]int)
+	rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
I don't think seeding from within the function body is good practice, the seed should be performed externally during initialisation. This means if we later decide that we want to re simulate with the same seed we are able to do so, and we can also log that seed. This should be resolved throughout the PR

Reply:
Every time an election take place, the number of ballots will be randomly generated. If we want to run that election once again with the ballot number same as last time, we should record details about past few elections (at least the last one). Do you mean this?

Reply:
When you call `rand.Seed()` you are causing a global state change because you are changing the random number sequence used by all code that uses `rand`. That's okay, but typically that is done by initialization code, and not in the specific function that uses `rand` like done here.

The problem with this approach is that every time this function is called the random number generator is re-seeded. Typically you only want to seed it once

Furthermore, sometimes you want to use a fixed seed. Say for example we wanted to run the exact same simulation twice, in order to do this we would use a fixed value for the seed in the initialization code. The current problem is, when this function is invoked, it will override the random seed making it impossible for us to do that

Long story short, we should call `rand.Seed()` once from main, but not from here (or any of these functions)